packages/nimbus/src/components/tooltip/tooltip-migration-guide.mdx

```diff
@@ -0,0 +1,176 @@
+---
+id: Components-tooltip-migration
+title: Tooltip Migration Guideline
+description: Engineering guideline for migrating from UI Kit to Nimbus Tooltip component
+documentState: InitialDraft
+order: 999
+menu:
+  - Components
+  - Feedback
+  - TooltipMigration
+tags:
+  - component
+  - engineering
+  - migration
+---
+
+# Tooltip Migration Guideline (Uikit to Nimbus)
+
+## Basic requirement
+
+Migration requirements like ensuring the right codemod exists, where to find it or put it etc. 
+
+
+### Code Mod Instructions
+
+If a code mod exists for Tooltip migration, run:
+
+```bash
+npx nimbus-codemods migrate-tooltip
+```
+
+### Manual Migration Steps
+
+For manual migration, follow these steps:
+
+#### Step 1: Replace Component Structure
+**Before (UI Kit):**
+```tsx
+<Tooltip title="Helpful information" isOpen={isOpen} onClose={handleClose}>
+  <Button>Hover me</Button>
+</Tooltip>
+```
+
+**After (Nimbus):**
+```tsx
+<Tooltip.Root isOpen={isOpen} onOpenChange={setIsOpen}>
+  <Button>Hover me</Button>
+  <Tooltip.Content>Helpful information</Tooltip.Content>
+</Tooltip.Root>
+```
+
+#### Step 2: Update Event Handlers
+**Before:**
+```tsx
+const handleClose = () => {
+  setIsOpen(false);
+};
+```
+
+**After:**
+```tsx
+const handleOpenChange = (isOpen: boolean) => {
+  setIsOpen(isOpen);
+};
+```
+
+#### Step 3: Handle Removed Props
+For `horizontalConstraint`, wrap with a container or use CSS:
+
+**Before:**
+```tsx
+<Tooltip title="Long text content..." horizontalConstraint="m">
+  <Button>Button</Button>
+</Tooltip>
+```
+
+**After:**
+```tsx
+<Tooltip.Root>
+  <Button>Button</Button>
+  <Tooltip.Content style={{ maxWidth: '200px' }}>
```

Comment:
Forgive my ignorance, isn't one of the main benefits of Nimbus that we can put styles directly on our components like `maxWidth="200px"`?

Reply:
No, forgive my ignorance actually 😅
You are correct!